### DISCUSSION TITLE
Uninstalling packages

### DISCUSSION THREAD

**Discussion by rmnprt** [pkgxdev/pkgx#617] [Help] [answered]
So I do `tea -S php` to install PHP.
How do I uninstall this?

I understand the [document](https://docs.tea.xyz/appendix/faq) states –
> Step into ~/.tea and delete the packages you no longer want. It’s that simple.

But how can I possibly know and delete all its dependencies? What am I missing?

**Comment by mxcl** [comment, marked as answer]
My advice would be to delete everything in `~/.tea` bar `tea.xyz`. `tea` is nice in that deleting stuff doesn’t effect workflows. Everything just gets installed again as you need it.

-----

We haven’t got something like `apt auto-remove` yet, nor a `deps` command. There is no easy way to recursively list deps with `tea/cli` currently.

You can `tea +php.net` to get the env dump and via that figure out all deps, but it's not nice.

We have something like a `tea install` command planned which moves us closer to normal package management, that may fit your needs here.

**Reply by rmnprt** [reply]
Takes a bit to get used to, but I'm starting to get the hang of how it presents as more of a global proxy binary than a package manager. Also good to know some of the conventional package manager commands are on the way. Thanks

**Reply by jhheider** [reply]
Untested, but approximations:
```shell
# Global clean up
$ find $(tea --prefix) -mindepth 1 -maxdepth 1 ! -name tea.xyz -exec rm -rf {} \;

# Only clean up php + deps
$ tea -E +php.net | grep ^PATH | tr : '\n' | grep "$(tea --prefix)" | sed -e "s_.*$(tea --prefix)/\(.*\)/bin_\1_" | tr '\n' '\0' | xargs -0 rm -rf
```

(Not claiming those are expressly useful, but felt like working them up just in case.)
